generate_index_html.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import re
+import subprocess
+import argparse
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+def get_custom_released_package_versions(bucket: str, folder: str) -> dict:
+    data = subprocess.check_output([f"gsutil ls -lh 'gs://{bucket}/{folder}/' | grep -E '\.whl|\.tar\.gz'"], shell=True)
+    # Currently similar strings are correctly parsed:
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2+astro+1-py3-none-any.whl
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.tar.gz
+    # - apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.whl
+    # Note - platform information is not handled atm.
+    pattern = re.compile("([\w\W]+?)  ([\w\W]+?)  gs://([\w\W]+?(?=/))/([\w\W]+?(?=/))/(([\w\W]+?)-(\d+\.\d+\.\d+(rc\d+)?(\+astro\.\d+)?)-py3-none-any(?:\.whl|\.tar\.gz))")
+    matches = pattern.findall(data.decode())
+    version_to_files = {}
+    for match in matches:
+        if match[6] not in version_to_files:
+            version_to_files[match[6]] = []
+        version_to_files[match[6]].append((match[1], match[4]))
+    return version_to_files
+
+
+def get_released_public_package_list(package: str) -> dict:
+    url = f"https://pypi.org/pypi/{package}/json"
+    response = requests.get(url)
+    response.raise_for_status()
+    return {release: files for release, files in response.json()["releases"].items()}
+
+
+def get_custom_package_json(package_info: tuple) -> dict:
+    """Generate dict data for a custom package"""
+    return {
+        "size": "45630",
```

Comment:
Why the constant 45630?

Reply:
The size value was not that important from the pip index perspective. But now it will be showing the actual file size. PTAL.